internal/provider/resource_gitlab_project.go

```diff
@@ -1137,3 +1144,7 @@ func flattenProjectPushRules(pushRules *gitlab.ProjectPushRules) (values []map[s
 		},
 	}
 }
+
+func namespaceOrPathChanged(ctx context.Context, d *schema.ResourceDiff, meta interface{}) bool {
+	return d.HasChange("namespace_id") || d.HasChange("path")
```

Comment:
Isn't the `name` argument itself also affected by the problem at hand here?

Reply:
Hmm, on the first thought I would say no, as the resource doesn't return a `name_with_namespace` attribute which is directly affected by a change of `name`.

But you're right, as `name` is the only required attribute and not `path` which can be derived from the `name`. I always explicitly set the `path` so it didn't come up on the tests..

Reply:
Meeeh, I can't find a way to cover that case if `path` isn't set and to trigger on the `name`. I don't just want to trigger on `name` generally as it may not change the depending attributes always. But a change on that e.g. may trigger recreating a dependent project on that `import_url`.

Reply:
@timofurrer Meeeh, I can't find a way to cover that case if `path` isn't set and to trigger on the `name`. I don't just want to trigger on `name` generally as it may not always change the depending attributes. But a change on that e.g. may trigger recreating a dependent project on that `import_url`.

Reply:
> Yes, you are right - it's only during creation, when path isn't set, right? Or did you uncover another situation?

... for which it actually isn't needed anyways 🤔

Reply:
Right, `CustomizeDiff` is applicable to updates not creation.

Reply:
> Yes, you are right - it's only during creation, when `path` isn't set, right? Or did you uncover another situation?

Ah, yes. `name` is only used as `path` during creation, when no `path` is send to the API. Afterwards `name` changes only with an unset `path` don't trigger changes in the repository path.